src/System.Reflection/tests/PropertyInfo/PropertyInfo_GetSetValueTests.cs

```diff
@@ -0,0 +1,374 @@
+            PropertyInfo pi = GetProperty(type, propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            Assert.Throws<ArgumentException>(() =>
+            {
+                var value = pi.GetValue(obj, testObj);
+            });
+        }
+
+
+        //Verify PropertyInfo.GetValue throws Exception
+        [Fact]
+        public static void TestGet_ThrowsException()
+        {
+            string propertyName = "PropertyC";
+            Object obj = Activator.CreateInstance(typeof(MyCoVariantTest));
+            PropertyInfo pi = GetProperty(typeof(MyCoVariantTest), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            // In Win8p instead of TargetException , generic Exception is thrown
+            // Refer http://msdn.microsoft.com/en-us/library/b05d59ty.aspx
+            Assert.ThrowsAny<Exception>(() => pi.GetValue(null, new Object[] { "1", "2" }));
+        }
+
+
+
+        [Theory]
+        [InlineData("NoSetterProp", 100, null)]
+        [InlineData("Item", "pw", new Object[] { 99, 2, "hello", "SOME  string" })]
+        [InlineData("Item", 100, new Object[] { 99, 2, new string[] { "hello" }, "SOME  string" })]
+        public static void TestSet_ThrowsArgumentException(string propertyName, Object setValue, Object[] index)
+        {
+            TypeWithProperties obj = new TypeWithProperties();
+            PropertyInfo pi = GetProperty(typeof(TypeWithProperties), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            Assert.Throws<ArgumentException>(() =>
+            {
+                pi.SetValue(obj, setValue, index);
+            });
+        }
+        
+        //Try to set instance Property with null object 
+        [Theory]
+        [InlineData("HasSetterProp", null, null)]
+        [InlineData("Item", "pw", new Object[] { 99, 2, "SOME string" })]
+        public static void TestSetNull_ThrowsException(string propertyName, Object setValue, Object[] index)
+        {
+            PropertyInfo pi = GetProperty(typeof(TypeWithProperties), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            // Generic Exception is thrown in WP8 instead of TargetException
+            Assert.ThrowsAny<Exception>(() => pi.SetValue(null, null, null));
+        }
+
+        //Try to set instance Property with wrong type
+        [Fact]
+        public static void TestSet_ThrowsException()
+        {
+            string propertyName = "HasSetterProp";
+            TypeWithProperties obj = new TypeWithProperties();
+            PropertyInfo pi = GetProperty(typeof(TypeWithProperties), propertyName);
+
+            Assert.Equal(pi.Name, propertyName);
+
+            // Generic Exception is thrown in WP8 instead of TargetException
+            Assert.ThrowsAny<Exception>(() => pi.SetValue(obj, "foo", null));
+        }
+        
+
+        //Try to set indexer Property , Incorrect number of parameters
```

Comment:
If we make the test names good enough, we shouldn't need to keep comments like this :) e.g.
```C#
public static void IndexerProperty_Set_IncorrectNumberOfParameters_ThrowsTargetParameterCountException()
```
or some such thing.